crates/sui-types/src/unit_tests/messages_tests.rs

```diff
@@ -45,14 +43,15 @@ fn test_signed_values() {
         /* voting right */ 0,
     );
     let committee = Committee::new_for_testing_with_normalized_voting_power(0, authorities);
-
+    let gas_price = 10;
     let transaction = Transaction::from_data_and_signer(
-        TransactionData::new_transfer_with_dummy_gas_price(
+        TransactionData::new_transfer(
             _a2,
             random_object_ref(),
             a_sender,
             random_object_ref(),
-            MAX_GAS_BUDGET_FOR_TESTING,
+            TEST_ONLY_GAS_UNIT_FOR_TRANSFER * gas_price,
```

Comment:
Think we need to make sure that we are always checking if budget set is  50SUI max and if we are just using 50SUI max for budget that the coin used also has >50 SUi. Think this will allow us to have the unified gas price everywhere of 1000.

Reply:
yeah we can add a `max` check in some of the transaction builder functions